machine-learning/05-machine_learning_exercise.Rmd

```diff
@@ -0,0 +1,245 @@
+
+We're going to look specifically at **pathways that change with treatment**, 
+but we precomputed the other objects in case you would like to understand what 
+samples tend to group together or what is driving the variation in the data.
+
+We'll use `dplyr` and `ggplot2` quite a bit in this notebook, so use the chunk
+below to load them.
+
+```{r}
+
+```
+
+The metadata is available at `data/data/GSE116436/metadata_GSE116436_cleaned.tsv`.
+Read it in with `readr` and save it as `metadata_df`
+
+```{r}
+metadata_df <- 
+```
+
+What metadata do we have for these samples ? 
+Use this chunk to explore `metadata_df`.
+
+```{r}
+
+```
+
+## What pathways are perturbed by the drugs in this experiment?
+
+The PLIER model we've precomputed is available `models/NCI60_PLIER_model.RDS`.
+Read it in with `readr` and save it as `plier_results`.
+
+```{r}
+plier_results <-
+```
+
+The `summary` element of what is returned by PLIER can tell us about
+associations, you can access it with `$` and it is a `data.frame`.
+
+Use this chunk to view the PLIER model `summary`.
+
+```{r}
+
+```
+
+Save the summary information as it's own data.frame called `summary_df`. 
+We'll use this object to find out more about our model. 
+
+```{r}
+summary_df <-
+```
+
+The latent variables from this model will be associated with a pathway if the
+gene expression data supports that there is some variation in the expression
+levels of genes in that pathway.
+It could be due to treatment with a drug or variation could arise from tissue- 
+or cell line-specific differences.
+
+### Is there a latent variable associated with the proteasome pathway?
+
+We can use a chain of `dplyr` functions to filter `summary_df`.
+Let's call `FDR < 0.05` associated; this is the same cutoff we used in the
+PLIER notebook earlier.
+
+```{r}
+summary_df %>%
+  dplyr::filter(,  # significant associations only
+                # grepl returns TRUE when the pattern "proteasome" is found
+                # in the pathway column
+                grepl("proteasome", pathway, ignore.case = TRUE))
+```
+
+The LV index column of `summary_df` tells us _which row_ in the LV by sample
+matrix `plier_results$B` we should look at.
+To make plotting easier downstream, let's make a data.frame that contains two 
+columns, `geo_accession` and `proteasome_lv` that contain the sample identifiers
+and the values for the latent variable associated with the proteasome gene set, 
+respectively.
+Recall that the column names of `plier_results$B` are the sample identifiers.
+(Hint: `?colnames`)
+
+Call this data.frame `proteasome_df`.
+
+```{r}
+proteasome_df <- data.frame(geo_accession = ,
+                            proteasome_lv = )
+```
+
+Now that you have `proteasome_df`, let's join it with our metadata using
+an `inner_join` from `dplyr` to form a new data.frame called `prot_meta_df`.
+We'll join by the sample names, which is what `by = "geo_accession"` specifies.
+
+```{r}
+prot_meta_df <- dplyr::inner_join(x = , 
```

Comment:
Could use <THESE> here. And other places.

Reply:
I don't follow

Reply:
Forgot that `<>` gets made invisible.